src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/SynthesizedClosureEnvironment.cs

```diff
@@ -127,9 +127,9 @@ internal override IEnumerable<FieldSymbol> GetFieldsToEmit()
         public override Symbol ContainingSymbol => _topLevelMethod.ContainingSymbol;
 
         // Closures in the same method share the same SynthesizedClosureEnvironment. We must
-        // always return true because two closures in the same method might have different
-        // AreLocalsZeroed flags.
-        public sealed override bool AreLocalsZeroed => true;
+        // never delegate to _topLevelMethod or OriginalContainingMethodOpt because two closures
+        // in the same method might have different AreLocalsZeroed flags.
+        public sealed override bool AreLocalsZeroed => ContainingType.AreLocalsZeroed;
```

Comment:
>ContainingType [](start = 55, length = 14)

It would be good to explain what the ```ContainingType``` is in this case and why does it makes sense to delegate to it, but never delegate to the ```_topLevelMethod```, which might have more "accurate" value.

Reply:
Actually, I just realized it does not make sense to delegate it to the type. We would have exactly the same problem (two closures with different AreLocalsZeroed flags sharing the same SynthesizedClosureEnvironment). I'll revert this change and keep the previous behavior.